openssl@1.1 install fails with missing perl lib

# Bug report

**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] ran `brew update` and can still reproduce the problem?
- [ ] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [ ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

<!-- To help us debug your issue, please complete these sections: -->

## What you were trying to do (and why)
Update `openssl@1.1`

## What happened (include command output)
https://gist.github.com/lewis6991/c23bbcc2a8611a843658b3a9efd49600
```bash
❯ brew upgrade openssl@1.1
==> Upgrading 1 outdated package:
openssl@1.1 1.1.1g -> 1.1.1h
==> Upgrading openssl@1.1 1.1.1g -> 1.1.1h
==> Downloading https://curl.haxx.se/ca/cacert-2020-01-01.pem
Already downloaded: /devtools/.cache/downloads/e2b36a2c425753a7b9d6a8eccb44b33cb7ed93cb91f96dafcf35fa18d8e0513a--cacert-2020-01-01.pem
==> Downloading https://cpan.metacpan.org/authors/id/L/LE/LEONT/Test-Harness-3.42.tar.gz
Already downloaded: /devtools/.cache/downloads/4a5728e2132a470084886ad6e6278c6b23584e314ee60dcd57ea679202d5da4a--Test-Harness-3.42.tar.gz
==> Downloading https://cpan.metacpan.org/authors/id/E/EX/EXODIST/Test-Simple-1.302175.tar.gz
Already downloaded: /devtools/.cache/downloads/af880136321fd9982d8eb9dd29977fec40c093fb2fd48d6fff85de8d6bb9a5b5--Test-Simple-1.302175.tar.gz
==> Downloading https://www.openssl.org/source/openssl-1.1.1h.tar.gz
Already downloaded: /devtools/.cache/downloads/074582038ae735e83b4ab0ea44ac5b654fa5502b7aadbcb01237403ac0878ceb--openssl-1.1.1h.tar.gz
==> perl Makefile.PL INSTALL_BASE=/devtools/linuxbrew/Cellar/openssl@1.1/1.1.1h/libexec
Last 15 lines from /scratch/lewrus01/.cache/Homebrew/Logs/openssl@1.1/01.perl:
2020-10-12 13:42:05 +0100

perl
Makefile.PL
INSTALL_BASE=/devtools/linuxbrew/Cellar/openssl@1.1/1.1.1h/libexec

Can't locate ExtUtils/MakeMaker.pm in @INC (@INC contains: /devtools/linuxbrew/Cellar/openssl@1.1/1.1.1h/libexec/lib/perl5 /usr/local/lib64/perl5 /usr/local/share/perl5 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5 .) at Makefile.PL line 8.
BEGIN failed--compilation aborted at Makefile.PL line 8.

READ THIS: https://docs.brew.sh/Troubleshooting
```

## What you expected to happen

To not error

## Step-by-step reproduction instructions (by running `brew install` commands)

System perl is 5.16.3
OS: RHE7

`brew upgrade openssl@1.1`

I have got the same error while install brew the first time under my home directory.

```
install libcrypto.so -> /home/zhuj6/.linuxbrew/Cellar/openssl@1.1/1.1.1h/lib/libcrypto.so
install libcrypto.so.1.1 -> /home/zhuj6/.linuxbrew/Cellar/openssl@1.1/1.1.1h/lib/libcrypto.so
ar: /home/zhuj6/.linuxbrew/Cellar/openssl@1.1/1.1.1h/lib/libcrypto.so: file format not recognized
ar: /home/zhuj6/.linuxbrew/Cellar/openssl@1.1/1.1.1h/lib/libcrypto.so.new: file format not recognized
make: *** [install_dev] Error 1

READ THIS: https://docs.brew.sh/Troubleshooting

These open issues may also help:
openssl@1.1 install fails with missing perl lib https://github.com/Homebrew/linuxbrew-core/issues/21252
```

Tried `brew install perl`, got the same error.

--- a/Formula/openssl@1.1.rb
+++ b/Formula/openssl@1.1.rb
@@ -39,6 +39,12 @@ class OpensslAT11 < Formula
       url "https://cpan.metacpan.org/authors/id/E/EX/EXODIST/Test-Simple-1.302175.tar.gz"
       sha256 "c8c8f5c51ad6d7a858c3b61b8b658d8e789d3da5d300065df0633875b0075e49"
     end
+
+    resource "ExtUtils::MakeMaker" do
+      url "https://cpan.metacpan.org/authors/id/B/BI/BINGOS/ExtUtils-MakeMaker-7.48.tar.gz"
+      sha256 "94e64a630fc37e80c0ca02480dccfa5f2f4ca4b0dd4eeecc1d65acd321c68289"
+    end
+
   end
 
   # SSLv2 died with 1.1.0, so no-ssl2 no longer required.
@@ -60,6 +66,13 @@ class OpensslAT11 < Formula
   def install
     unless OS.mac?
       ENV.prepend_create_path "PERL5LIB", libexec/"lib/perl5"
+
+      resource("ExtUtils::MakeMaker").stage do
+        system "perl", "Makefile.PL", "INSTALL_BASE=#{libexec}"
+        system "make", "PERL5LIB=#{ENV["PERL5LIB"]}", "CC=#{ENV.cc}"
+        system "make", "install"
+      end
+
       resource("Test::Harness").stage do
         system "perl", "Makefile.PL", "INSTALL_BASE=#{libexec}"
         system "make", "PERL5LIB=#{ENV["PERL5LIB"]}", "CC=#{ENV.cc}"

--- a/Formula/openssl@1.1.rb
+++ b/Formula/openssl@1.1.rb
@@ -67,22 +67,12 @@ class OpensslAT11 < Formula
     unless OS.mac?
       ENV.prepend_create_path "PERL5LIB", libexec/"lib/perl5"
 
-      resource("ExtUtils::MakeMaker").stage do
-        system "perl", "Makefile.PL", "INSTALL_BASE=#{libexec}"
-        system "make", "PERL5LIB=#{ENV["PERL5LIB"]}", "CC=#{ENV.cc}"
-        system "make", "install"
-      end
-
-      resource("Test::Harness").stage do
-        system "perl", "Makefile.PL", "INSTALL_BASE=#{libexec}"
-        system "make", "PERL5LIB=#{ENV["PERL5LIB"]}", "CC=#{ENV.cc}"
-        system "make", "install"
-      end
-
-      resource("Test::More").stage do
-        system "perl", "Makefile.PL", "INSTALL_BASE=#{libexec}"
-        system "make", "PERL5LIB=#{ENV["PERL5LIB"]}", "CC=#{ENV.cc}"
-        system "make", "install"
+      %w[ExtUtils::MakeMaker Test::Harness Test::More].each do |r|
+        resource(r).stage do
+          system "perl", "Makefile.PL", "INSTALL_BASE=#{libexec}"
+          system "make", "PERL5LIB=#{ENV["PERL5LIB"]}", "CC=#{ENV.cc}"
+          system "make", "install"
+        end
       end
     end
 

